devnet-sdk/devstack/stack/match/gate.go

```diff
@@ -0,0 +1,31 @@
+package match
+
+import (
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/devstack/devtest"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/devstack/stack"
+)
+
+type skipIfNone[I comparable, E stack.Identifiable[I]] struct {
+	t     devtest.T
+	inner stack.Matcher[I, E]
+}
+
+func (s *skipIfNone[I, E]) Match(elems []E) []E {
+	elems = s.inner.Match(elems)
+	s.t.Gate().NotEmpty(elems, "must match something to continue, but matched nothing with %s", s.inner)
+	return elems
+}
+
+func (s *skipIfNone[I, E]) String() string {
+	return fmt.Sprintf("SkipIfNone(%s)", s.inner)
+}
+
+// SkipIfNone skips the test if no elements were matched with the inner matcher
+func SkipIfNone[I comparable, E stack.Identifiable[I]](t devtest.T, inner stack.Matcher[I, E]) stack.Matcher[I, E] {
```

Comment:
Not a blocking thing, but I wonder if there's a better name that makes it clearer that we're gating on this being required rather than specifically that we're skipping (particularly given the comment above about potentially wanting to fail if preconditions are met).  JUnit uses `assume` (vs `assert`) for that, so maybe `AssumeMatch`?

If we stuck with the `Gate` type language it could be `GateOnMatch`.

Reply:
It's not go-like to repeat the package name in the method name. I think `match.Assume(<thing>)` makes sense though. I'll rename it.

Reply:
Ah yeah, hard to see the package name context in the GitHub diff - `match.Assume` is nice.